lib/teams/aws-batch/aws-batch-on-eks-team.ts

```diff
@@ -0,0 +1,167 @@
+import { assertEC2NodeGroup } from "../../cluster-providers";
+import { ClusterInfo, Values } from "../../spi";
+import { ApplicationTeam, TeamProps } from "../team";
+import { KubectlProvider, ManifestDeployment } from "../../addons/helm-addon/kubectl-provider";
+import { loadYaml, readYamlDocument } from "../../utils/yaml-utils";
+import { Construct } from "constructs";
+import { createNamespace } from "../../utils";
+
+import * as batch from 'aws-cdk-lib/aws-batch';
+import * as eks from 'aws-cdk-lib/aws-eks';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { AutoScalingGroup } from "aws-cdk-lib/aws-autoscaling";
+import * as iam from "aws-cdk-lib/aws-iam";
+
+const NAMESPACE = 'aws-batch';
+
+/**
+ * Interface to define an AWS Batch on EKS team
+ */
+export interface BatchEksTeamProps extends TeamProps {
+  /**
+   * Name of the team
+   */
+  name: string,
+
+  /**
+   * Namespace name for AWS Batch
+   */
+  namespace: string
+
+  /**
+   * Allocation strategies for EKS Compute environment
+   */
+  allocationStrategy?: 'BEST_FIT_PROGRESSIVE' | 'SPOT_CAPACITY_OPTIMIZED'
+
+  jobQueueName: string
+
+  priority: number,
```

Comment:
what could it be? Can be negative or 1 billion? Is there a range?

Reply:
whole number. i have not tested the limits of what it could be (and it's certainly something not documented anywhere). I'll put in some assert statements along the way.